turbo/generators/config.ts

```diff
@@ -0,0 +1,131 @@
+import { PlopTypes } from '@turbo/gen';
+
+// Convert a string to kebab-case
+function kebabCase(str) {
+  return str
+    // Replace lowercase letter followed by an uppercase letter with a hyphen in between
+    .replace(/([a-z])([A-Z])/g, '$1-$2')
+    // Replace whitespace and underscores with hyphens
+    .replace(/[\s_]+/g, '-')
+    // Convert the string to lowercase
+    .toLowerCase();
+}
+
+function validateNotEmpty(input:string) {
+  if (!input.length) {
+    return 'This field cannot be left empty!';
+  }
+  return true;
+}
+
+export default function generator(plop: PlopTypes.NodePlopAPI): void {
+  /**
+   * custom action to log error when the package name length is long.
+   * 'error': Name of the action
+   */
+  plop.setActionType('error', (config) => {
+    const { name } = (config as { name: string });
+    return Promise.reject(Error(`${name} is too long for the package, please shorten it to 128 character or below.`));
+  });
+
+  plop.setGenerator('packages', {
+    description: 'Generate a base config for new package',
+    prompts: [
+      {
+        type: 'list',
+        name: 'library',
+        message: 'Select a library the package will belong to:',
+        choices: ['ecc'],
+      },
+      {
+        type: 'list',
+        name: 'libraryDomain',
+        message: 'Select domain of the package:',
+        choices: ['client', { name: 'Fill custom input for this field', value: '__custom__' }],
```

Comment:
Customs is not a option!

Reply:
@git-anurag-hub Sorry, I couldn't find it in their (PlopJs) documentation. If there is a better way to accomplish the task or if I overlooked any relevant details, please let me know, and I'll be more than happy to update my response accordingly.

Reply:
Ohh what I meant to say is that there will be only one choice for the domain. But after the messages in the slack now there will be only two domain for any package i.e. client, utils! Don't give the user the option to have custom domain name.

Reply:
ok! I thought you were asking me to refactor the code, Also `utils`? this domain name hasn't been discussed afaik also because if that's the case `ecc-ui-utils` will need to be renamed too to `ecc-utils-ui` right ?,  I'll make the changes ASAP though, only client and utils will be part of domain name as of now :+1:

Reply:
Thanks @JaeAeich